homeassistant/components/image/__init__.py

```diff
@@ -146,8 +151,48 @@ def image(self) -> bytes | None:
 
     async def async_image(self) -> bytes | None:
         """Return bytes of image."""
+
+        async def _async_load_image_from_url(url: str) -> Image | None:
+            """Load an image by url."""
+            try:
+                response = await self._client.get(
+                    url, timeout=GET_IMAGE_TIMEOUT, follow_redirects=True
+                )
+                response.raise_for_status()
+                return Image(
+                    content=response.content,
+                    content_type=response.headers["content-type"],
+                )
+            except httpx.TimeoutException:
+                _LOGGER.error("%s: Timeout getting image from %s", self.entity_id, url)
+                return None
+            except (httpx.RequestError, httpx.HTTPStatusError) as err:
+                _LOGGER.error(
+                    "%s: Error getting new image from %s: %s",
+                    self.entity_id,
+                    url,
+                    err,
+                )
+                return None
+
+        if (url := await self.async_image_url()) is not None:
+            # Ignore an empty url
+            if url == "":
+                return None
+            if (image := await _async_load_image_from_url(url)) is None:
+                return None
+            self._attr_content_type = image.content_type
+            return image.content
         return await self.hass.async_add_executor_job(self.image)
 
+    def image_url(self) -> str | None:
+        """Return URL of image."""
+        return None
+
+    async def async_image_url(self) -> str | None:
+        """Return URL of image."""
+        return self.image_url()
```

Comment:
If this method needs to be able to await we should expect the sync version of the method to be blocking. Then we should schedule the work on the executor.